### PR TITLE
[Merged by Bors] - refactor(CategoryTheory/Sites): continuous functors

### PR DESCRIPTION
**Pull request by joelriou**
This PR introduces the typeclass `Functor.IsContinuous` which says that the precomposition with a functor preserves the sheaf condition for Grothendieck topologies. It slightly refactors the previous main theorem about `CoverPreserving` and `CompatiblePreserving` functors: it now states that such functors are continuous. The *pushforward* functor for a continuous functor is defined under the `Functor.IsContinuous` assumption rather than the combination of both `CoverPreserving` and `CompatiblePreserving`. The property `CoverLifting` is renamed `IsCocontinuous` and it is made a class. The property `IsCoverDense` is also made a class. 

---

The current results are great, but I feel the terminology was very confusing: for example, the functor between categories of sheaves induced by a continuous functor was named `pullback`, even though it should be considered as a *pushforward* functor (for example, they induce the covariant functoriality of sheaves for sites or topological spaces). I have made some other similar renames.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Sheaves/SheafCondition/Sites.lean [hunk 1]
```diff
@@ -18,8 +18,8 @@ topology, in preparation of connecting the sheaf condition on sites to the vario
 on spaces.
 
 We also specialize results about sheaves on sites to sheaves on spaces; we show that the inclusion
-functor from a topological basis to `TopologicalSpace.Opens` is `CoverDense`, that open maps
-induce `CoverPreserving` functors, and that open embeddings induce `CompatiblePreserving` functors.
+functor from a topological basis to `TopologicalSpace.Opens` is cover dDense`, that open maps
```
Comment (alreadydone) [inline Mathlib/Topology/Sheaves/SheafCondition/Sites.lean:21]:
```suggestion
functor from a topological basis to `TopologicalSpace.Opens` is cover dense, that open maps
```

File: Mathlib/Topology/Sheaves/SheafCondition/Sites.lean [hunk 2]
```diff
@@ -176,9 +176,18 @@ theorem IsOpenMap.coverPreserving (hf : IsOpenMap f) :
   exact ⟨_, hf.functor.map i, ⟨_, i, 𝟙 _, hV, rfl⟩, Set.mem_image_of_mem f hxV⟩
 #align is_open_map.cover_preserving IsOpenMap.coverPreserving
 
+
+lemma OpenEmbedding.isContinuous (h : OpenEmbedding f) :
```
Comment (dagurtomas) [inline Mathlib/Topology/Sheaves/SheafCondition/Sites.lean:180]:
```suggestion
lemma OpenEmbedding.isContinuousFunctor (h : OpenEmbedding f) :
```
The name now sounds like it means the same as `OpenEmbedding.continuous`

**Review by dagurtomas — APPROVED** [review]

**Comment by dagurtomas** [comment]
I think that if we're introducing the name `Continuous` for functors, we should go all the way and add `Cocontinuous` as well. The docstring of `CoverLifting.lean` says: 

"This concept is also known as *cocontinuous functors* or
*cover-reflecting functors*, but we have chosen this name following [MM92] in order to avoid
potential naming collision or confusion with the general definition of cocontinuous functors
between categories as functors preserving small colimits."

This applies to the name *continuous functor* as well, but since we don't use that terminology for (co)limit preserving functors in mathlib, I think we can just remove that paragraph and claim both names for the sheaf-theoretic notion (and maybe add a warning that they don't mean "(co)limit-preserving".

**Comment by dagurtomas** [comment]
Could you also change line 44 in `CoverPreserving.lean` to `* https://stacks.math.columbia.edu/tag/00WU` to cite the whole section in Stacks and not just one particular lemma?

**Comment by joelriou** [comment]
Thanks for the suggestions @dagurtomas! Now, there is `Functor.IsContinuous` and `Functor.IsCocontinuous`.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Comment by joelriou** [comment]
retrying

bors merge

**Pull request closed without merging**